### PR TITLE
Reconcile dev after merge to main for v1.45.1

### PR DESCRIPTION
**Pull request by Geal**
Follow-up to the v1.45.1 being officially released, bringing version bumps and changelog updates into the dev branch.

Separating this from #5036 to solve the conflicts between main and dev without making an additional commit to main

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [ ] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

File: apollo-router/src/spec/query.rs [hunk 1]
```diff
@@ -298,8 +298,8 @@ impl Query {
         tracing::trace!(?recursion_limit, "recursion limit data");
 
         let hash = QueryHashVisitor::hash_query(
-            api_schema_definitions,
-            &api_schema.raw_sdl,
+            schema.supergraph_schema(),
+            &schema.raw_sdl,
```
Comment (Geal) [inline apollo-router/src/spec/query.rs:302]: same as the other note, this can be moved to a separate PR

**Review by o0Ignition0o — APPROVED** [review]

**Review by bnjjj — APPROVED** [review]

File: apollo-router/src/query_planner/bridge_query_planner.rs [hunk 2]
```diff
@@ -835,21 +831,20 @@ impl Service<QueryPlannerRequest> for BridgeQueryPlanner {
...
                     let executable_document = modified_query
-                        .to_executable_validate(api_schema_definitions)
+                        .to_executable_validate(api_schema)
                         // Assume transformation creates a valid document: ignore conversion errors
                         .map_err(|e| SpecError::ValidationError(e.into()))?;
                     let hash = QueryHashVisitor::hash_query(
-                        api_schema_definitions,
-                        &api_schema.raw_sdl,
+                        this.schema.supergraph_schema(),
+                        &this.schema.raw_sdl,
```
[the remark this answers is not in the document]
Comment (abernix) [reply apollo-router/src/query_planner/bridge_query_planner.rs:847]: I just want to clarify that while this is an additional fix necessary for the re-introduction of https://github.com/apollographql/router/pull/5028, it **does not** constitute a necessary fix _right now_ given that https://github.com/apollographql/router/pull/5028 was exhaustive enough in and of itself.

**Review by abernix — COMMENTED** [review]
Should we land this and track the follow-up somewhere else?

**Comment by Geal** [comment]
we can land this. There was one additional fix I had to do to make tests pass: [5f86cd4](https://github.com/apollographql/router/pull/5037/commits/5f86cd454caa87f7dc20749d44c6ada93ddc0bcf)

**Comment by Geal** [comment]
[50eb674](https://github.com/apollographql/router/pull/5037/commits/50eb674c7686b12d74e6ccfa83d5310d47eab86f) is due to the typed conditions PR being merged in dev while 1.45.1 was worked on, so the schema aware hashes for subgraph queries change for those tests because this PR changes the algorithm with the fixes from 1.45.1
